src/main/java/se/liu/ida/hefquin/engine/query/impl/BGPImpl.java

```diff
@@ -30,7 +32,19 @@ public boolean equals( final Object o ) {
 			otps = ((BGPImpl) o).tps;
 		else
 			otps = ((BGP) o).getTriplePatterns();
-		if ( tps == otps )
+		
+		final Set<Triple> tpsTriple = new HashSet<>();
+		final Set<Triple> otpsTriple = new HashSet<>();
+		
+		for(TriplePattern i : tps) {
+			tpsTriple.add(i.asJenaTriple());
+		}
+		
+		for(TriplePattern j : otps) {
+			otpsTriple.add(j.asJenaTriple());
+		}
+		
+		if ( tpsTriple.equals(otpsTriple) )
 			return true;
 		else if ( tps.size() != otps.size() )
 			return false;
```

Comment:
this can be simplified:
```suggestion
		return tpsTriple.equals(otpsTriple)
```